exec/java-exec/src/test/java/org/apache/drill/exec/planner/common/TestNumericEquiDepthHistogram.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.planner.common;
+
+import org.apache.drill.categories.PlannerTest;
+
+import org.apache.drill.shaded.guava.com.google.common.collect.BoundType;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.Assert;
+import org.apache.drill.shaded.guava.com.google.common.collect.Range;
+
+
+@Category(PlannerTest.class)
+public class TestNumericEquiDepthHistogram {
+
+  @Test
+  public void testHistogramWithUniqueEndpoints() throws Exception {
+    int numBuckets = 10;
+    int numRowsPerBucket = 250;
+
+    // init array with numBuckets + 1 values
+    Double[] buckets = {1.0, 10.0, 20.0, 30.0, 40.0, 50.0, 60.0, 70.0, 80.0, 90.0, 100.0};
+
+    NumericEquiDepthHistogram histogram = new NumericEquiDepthHistogram(numBuckets);
+
+    for (int i = 0; i < buckets.length; i++) {
+      histogram.setBucketValue(i, buckets[i]);
+    }
+    histogram.setNumRowsPerBucket(numRowsPerBucket);
+
+    // Range: <= 1.0
```

Comment:
Do we want to account for the equality value e.g. 1.0 in this case? We already have the `ndv` so we can compute the number of rows for this value assuming uniform distribution.

Reply:
Done